src/services/services.ts

```diff
@@ -3903,7 +3903,8 @@ module ts {
                 }
 
                 // if this symbol is visible from its parent container, e.g. exported, then bail out
-                if (symbol.parent) {
+                // if symbol correspond to the union property - bail out
+                if (symbol.parent || (symbol.getFlags() & SymbolFlags.UnionProperty)) {

```

Comment:
I believe because union property consists of multiple components (where every component might have a parent) and it makes no sense to speak about a single parent (unless it will be something new to represent union of parents for all constituents)